staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -501,7 +501,6 @@ type VirtualMachineInstanceNetworkInterface struct {
 	// Hardware address of a Virtual Machine interface
 	MAC string `json:"mac,omitempty"`
 	// Name of the interface, corresponds to name of the network assigned to the interface
-	// TODO: remove omitempty, when api breaking changes are allowed
```

Comment:
> The comment on the API, about making the iface name mandatory is also
> removed, as this is known to be a valid option. Ifaces detected in the
> guest by the guest-agent, with no ability to link them to an existing
> network in the VMI spec will miss the name.

AFAIU the [code](https://github.com/kubevirt/kubevirt/blob/fc60a8edc53665350b6aac5efcd69eeb2200e40d/pkg/virt-handler/vm.go#L974), even those interfaces will have a name.